src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -901,3 +905,14 @@ public partial class WarningHeaderValue : System.ICloneable
         public static bool TryParse([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] string? input, [System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] out System.Net.Http.Headers.WarningHeaderValue? parsedValue) { throw null; }
     }
 }
+namespace System.Net.Http.Metrics
+{
+    public sealed class HttpMetricsEnrichmentContext
```

Comment:
Do we need to explicitly add something like
```c#
internal HttpMetricsEnrichmentContext() { }
```
to the ref and our code to prevent outside callers from constructing these?

Or was the intent for these to have public ctors?

Reply:
Good catch, the type should not be constructible by users and there was no public ctr in the [approved API](https://github.com/dotnet/runtime/issues/86281#issuecomment-1613667260).

```C#
internal HttpMetricsEnrichmentContext() { }
```

Do you know if this is this something we need to explicitly put into API proposals to signal that the ctr is hidden, or shall I just go ahead adding this without initiating further discussions?

Reply:
I believe it's usually at least mentioned during API review, but given that all the properties were approved as `get`-only and there is no other extensibility there, I don't see any value in the ctor being public.

> shall I just go ahead adding this without initiating further discussions?

I doubt anyone will complain :)